public/app/core/reducers/navModel.ts

```diff
@@ -4,10 +4,19 @@ import { cloneDeep } from 'lodash';
 import { NavIndex, NavModel, NavModelItem } from '@grafana/data';
 import config from 'app/core/config';
 
+export const HOME_NAV_ID = 'home';
+
 export function buildInitialState(): NavIndex {
   const navIndex: NavIndex = {};
   const rootNodes = cloneDeep(config.bootData.navTree as NavModelItem[]);
-  buildNavIndex(navIndex, rootNodes);
+  if (config.featureToggles.topnav) {
+    const homeNavIndex = rootNodes.findIndex((node) => node.id === HOME_NAV_ID);
+    const homeNav = rootNodes.splice(homeNavIndex, 1)[0];
+    buildNavIndex(navIndex, [homeNav]);
+    buildNavIndex(navIndex, rootNodes, homeNav);
+  } else {
+    buildNavIndex(navIndex, rootNodes);
+  }
```

Comment:
not sure I understand this change

Reply:
@torkelo `buildNavIndex` takes an optional 3rd parameter which is the `parentItem` for the nodes you pass in the second parameter. 

when `topnav` is off, it's the same as it always was.

when `topnav` is on, we filter out the home node from the other nodes. we build the navIndex for the home node first, then build the navIndex for all the other nodes in the `navTree` setting the home node as the parent for each of them.